packages/flutter/lib/src/foundation/_isolates_io.dart

```diff
@@ -13,56 +13,56 @@ import 'isolates.dart' as isolates;
 /// The dart:io implementation of [isolate.compute].
 Future<R> compute<Q, R>(isolates.ComputeCallback<Q, R> callback, Q message, { String? debugLabel }) async {
   debugLabel ??= kReleaseMode ? 'compute' : callback.toString();
+
   final Flow flow = Flow.begin();
   Timeline.startSync('$debugLabel: start', flow: flow);
-  final ReceivePort resultPort = ReceivePort();
-  final ReceivePort exitPort = ReceivePort();
-  final ReceivePort errorPort = ReceivePort();
+  final ReceivePort port = ReceivePort();
   Timeline.finishSync();
+
+  final Completer<dynamic> completer = Completer<dynamic>();
+  port.listen((dynamic msg) {
+    completer.complete(msg);
+
+    Timeline.startSync('$debugLabel: end', flow: Flow.end(flow.id));
+    port.close();
+    Timeline.finishSync();
+  });
+
```

Comment:
I would have done `port.first` instead, but the previous iteration of the code was timing the time it takes to close the ports and `port.first` does that implicitly. Thus, the use of `Completer` to maintain the original logic.

Reply:
Changed `ReceivePort` to `RawReceivePort` because of the limited API we are using